### PR TITLE
Add sparse architecture and test with torchrec SparseArch.

### PR DESCRIPTION
**Pull request by pashu123**
Features that don't work with current implementation:
    -- embeddingbag config with multiple features.

### CODE REVIEW

File: shark/examples/shark_inference/sparse_arch.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+            prev_values, prev_offsets = calculate_offsets(
+                dense_list, prev_values, prev_offsets
+            )
+        combined_list.append(prev_values)
+        combined_list.append(prev_offsets)
+
+    return combined_list
+
+
+class SparseArchNod(nn.Module):
```
Comment (powderluv) [inline shark/examples/shark_inference/sparse_arch.py:70]: minor nit: maybe use use SparseArchSHARK branding ?

**Review by powderluv — DISMISSED** [review]

File: shark/examples/shark_inference/sparse_arch.py (hunk 1 above)
Comment (pashu123) [reply shark/examples/shark_inference/sparse_arch.py:70]: Sounds good.

Comment (pashu123) [reply shark/examples/shark_inference/sparse_arch.py]: Done.

**Comment by pashu123** [comment]
@powderluv The tests pass from here: https://github.com/pytorch/torchrec/blob/71d51c8764c141ff8d849f73bcf06548ffad36c4/torchrec/models/tests/test_dlrm.py#L361

**Comment by pashu123** [comment]
I have added that as a test here.

**Review by powderluv — APPROVED** [review]

**Pull request merged**